src/pymortests/strategies.py

```diff
@@ -0,0 +1,451 @@
+
+
+def numpy_list_vector_spaces(draw, np_data_list, compatible, count, dims):
+    return [(NumpyListVectorSpace(d), ar) for d, ar in zip(dims, np_data_list)]
+
+
+def block_vector_spaces(draw, np_data_list, compatible, count, dims):
+    ret = []
+    rr = draw(hyst.randoms())
+
+    def _block_dims(d):
+        bd = []
+        while d > 1:
+            block_size = rr.randint(1, d)
+            bd.append(block_size)
+            d -= block_size
+        if d > 0:
+            bd.append(d)
+        return bd
+
+    for c, (d, ar) in enumerate(zip(dims, np_data_list)):
+        # only redraw after initial for (potentially) incompatible arrays
+        if c == 0 or (not compatible and c > 0):
+            block_dims = _block_dims(d)
+        constituent_spaces = [NumpyVectorSpace(dim) for dim in block_dims]
+        # TODO this needs to be relaxed again
+        assume(len(constituent_spaces))
+        ret.append((BlockVectorSpace(constituent_spaces), ar))
+    return ret
+
+_other_vector_space_types = []
+
+if config.HAVE_FENICS:
+    FENICS_spaces = {}
+    
+    def fenics_vector_spaces(draw, np_data_list, compatible, count, dims):
+        ret = []
+        for d, ar in zip(dims, np_data_list):
+            assume(d > 1)
+            if d not in FENICS_spaces:
+                FENICS_spaces[d] = FenicsVectorSpace(df.FunctionSpace(df.UnitIntervalMesh(d - 1), 'Lagrange', 1))
+            ret.append((FENICS_spaces[d], ar))
+        return ret
+    _other_vector_space_types.append('fenics')
+
+if config.HAVE_NGSOLVE:
+    def ngsolve_vector_spaces(draw, np_data_list, compatible, count, dims):
+        return [(create_ngsolve_space(d), ar) for d, ar in zip(dims, np_data_list)]
+    _other_vector_space_types.append('ngsolve')
+
+if config.HAVE_DEALII:
+    def dealii_vector_spaces(draw, np_data_list, compatible, count, dims):
+        return [(DealIIVectorSpace(d), ar) for d, ar in zip(dims, np_data_list)]
+    _other_vector_space_types.append('dealii')
+
+
+_picklable_vector_space_types = ['numpy', 'numpy_list', 'block']
+
+
+@hyst.composite
+def vector_arrays(draw, space_types, count=1, dtype=None, length=None, compatible=True):
+    dims = draw(hy_dims(count, compatible))
+    dtype = dtype or draw(hy_dtypes)
+    lngs = draw(length or hyst.tuples(*[hy_lengths for _ in range(count)]))
+    np_data_list = [draw(np_arrays(l, dim, dtype=dtype)) for l, dim in zip(lngs, dims)]
+    space_type = draw(hyst.sampled_from(space_types))
+    space_data = globals()[f'{space_type}_vector_spaces'](draw, np_data_list, compatible, count, dims)
+    ret = [sp.from_numpy(d) for sp, d in space_data]
+    assume(len(ret))
+    if len(ret) == 1:
+        assert count == 1
+        # in test funcs where we only need one array this saves a line to access the single list element
+        return ret[0]
+    assert count > 1
+    return ret
+
+
+def implementations(which='all', count=1, dtype=None, length=None, compatible=True, index_strategy=None, **kwargs):
```

Comment:
Sorry for bringing this up so late, but I think `implementations` is a bad name. We have implementations of all sorts, not only `VectorArrays`. Also, with `implementations`, I would assume to get classes, not instances. Maybe, a better name, which also shows that this uses hypotheses under the hood would be `given_vector_arrays` (or just `vector_arrays`)?

Reply:
Sure